homeassistant/components/xiaomi_miio/config_flow.py

```diff
@@ -32,6 +39,44 @@
 DEVICE_MODEL_CONFIG = {vol.Optional(CONF_MODEL): vol.In(MODELS_ALL)}
 
 
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        settings_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_CLOUD_SUBDEVICES,
```

Comment:
Why are these options part of an options flow? Normally connection details are part of the main config flow. Shouldn't we ask the user to choose what kind of connection to use when creating the entry?

Reply:
It is not really chosing, currently the local connection details are nessesary and suffiecient to set up the gateway.
The cloud credentials are optional and not needed, but the local connection details are always needed.

If cloud credentials are specified, the cloud is used to get the list of subdevices which can also be obtained by local control for most models of gateways, except for the EU gateway model (it lacks the required local command), the EU gateway itself can be controlled through local commands, only the list of subdevices cannot be optained (they can be controlled localy).
So the cloud credentials are currently an option to get subdevices for the EU gateway.

Reply:
Do all country gateway types support the cloud option or only the EU gateway? Do the other countries benefit from using the cloud option. Maybe we should only ask for the cloud option when the user configures an EU gateway?

Reply:
All gateway types do support the cloud option (that is the way the xiaomi miio home app works).

It is not really about the country but the model of the gateway (I call it EU gateway because that model has an EU power plug and it ment to be sold in the EU but obviously you can also use it outside the EU countries or use a diffrent model gateway inside the EU, the diffrent models are listed here https://www.home-assistant.io/integrations/xiaomi_miio/#supported-xiaomi-gateway-models).

At the moment other gateway models don't really benefit from using the cloud option, although I have seen some issues where localy retrieving the device list gives errors (related to max message size in miio local portecol) if large amounts of subdevices are connected (40+), users experiancing such problems cloud probably use the cloud option to get it working, but that has not been tested.
Besides for the future there are things that can be done with the cloud which is locally not possible (such as retrieving the last time (timestamp) a motion sensor has seen motion)